### PR TITLE
add more details to user-details processor

### PR DESCRIPTION
**Pull request by siddhantk232**
Add github specific information (id, name, username) to the `user-details` processor.
An example built using this version of `fastn` can be found at [siddhantk232/fastn-gh-auth-test](https://github.com/siddhantk232/fastn-gh-auth-test)

### CODE REVIEW

File: fastn-core/src/auth/github/apis.rs
Comment (amitu) [reply fastn-core/src/auth/github/apis.rs:144]: I am happy to merge this PR and do this in another PR.

Comment (siddhantk232) [reply fastn-core/src/auth/github/apis.rs:144]: you mean something like:

```rust
static USER_DETAILS_CACHE: once_cell::sync::Lazy<std::sync::Mutex<HashMap<String, GhUserDetails>>> =
    once_cell::sync::Lazy::new(|| std::sync::Mutex::new(HashMap::new()));

```

on login we add to this cache and on logout we remove from this cache?

Comment (amitu) [reply fastn-core/src/auth/github/apis.rs:144]: Not in-memory cache, that will be purged, some more persistent one, eg Postgres or Redis. What are you thinking of using as the cache key btw?

Comment (amitu) [reply fastn-core/src/auth/github/apis.rs:144]: I am thinking we should add session support, create a session id cookie on login, and store all session data on server side in db or redis against that session id.

Comment (siddhantk232) [reply fastn-core/src/auth/github/apis.rs:144]: Let's do this in another PR :slightly_smiling_face:

**Review by amitu — APPROVED** [review]
Looks good to me. Let's wait for @Arpita-Jaiswal's review also before merging.

File: fastn-core/src/auth/github/apis.rs
Comment (siddhantk232) [reply fastn-core/src/auth/github/apis.rs:144]: https://github.com/fastn-stack/fastn/issues/1487

**Review by Arpita-Jaiswal — APPROVED** [review]

File: fastn-core/src/library2022/processor/user_details.rs
Comment (AbrarNitk) [inline fastn-core/src/library2022/processor/user_details.rs:14]: It is better to use API as `fastn_core::auth::decrypt`, as an API user should not be worried about `utils`, and the argument in the function `decrypt_str` is going to be `str` always, do we really need `_str` in the method name?

File: fastn-core/src/library2022/processor/user_details.rs
Comment (AbrarNitk) [inline fastn-core/src/library2022/processor/user_details.rs:21]: `eprintln!` does not make sense here in anyways, use `format!` instead for formatting or use `logger` for logging.

File: fastn-core/src/library2022/processor/user_details.rs
Comment (AbrarNitk) [inline fastn-core/src/library2022/processor/user_details.rs:24]: `fastn_core::auth::github::apis::user_details` also can be reduced to `fastn_core::auth::github::user_details`.

File: fastn-core/src/library2022/processor/user_details.rs
Comment (AbrarNitk) [inline fastn-core/src/library2022/processor/user_details.rs:45]: You can also consider changing this `fastn_core::auth::github::apis::GhUserDetails` to `fastn_core::auth::github::GhUserDetails`. 

And do we need `Option<fastn_core::auth::github::api::GhUserDetails>`
or it can work without `Option`, because by looking at your code you are creating the value of `GhUserDetails` at only a single place and there you are using `Some(user)`.

**Comment by AbrarNitk** [comment]
@amitu please think of something for testing the feature, for me at least it does not make sense to test the feature in a different PR.

Let's say if `Abrar` wants to test this feature so I need to clone the different PR with current behavior or something like `fastn test gh-user <cookie>`.
- My assumptions are to consider `fbt` or [`criterion](https://github.com/bheisler/criterion.rs) for `cli` testing.

**Comment by amitu** [comment]
@AbrarNitk good ideas. @siddhantk232 keep testing also in mind. We have fbt tests, but you can write rust tests as well.

**Comment by amitu** [comment]
@siddhantk232 consider incorporating the feedback by @AbrarNitk after the merge.

File: fastn-core/src/library2022/processor/user_details.rs
Comment (siddhantk232) [reply fastn-core/src/library2022/processor/user_details.rs:45]: User  is an `Option<_>` because it's initialized to `None` which stays `None` if `github` cookie is not available or decrypting its value fails.

Comment (siddhantk232) [reply fastn-core/src/library2022/processor/user_details.rs:45]: I will change `fastn_core::auth::github::apis::GhUserDetails` to `fastn_core::auth::github::GhUserDetails`, thanks :+1:

**Comment by siddhantk232** [comment]
Thanks @AbrarNitk for the review. I'll add another PR to address these issues and will keep testing in mind as well :pray:

**Pull request merged**
